pkg/verifier/notaryv2/notaryv2.go

```diff
@@ -154,7 +154,7 @@ func getVerifierService(conf *NotaryV2VerifierConfig) (notation.Verifier, error)
 		certStores: conf.VerificationCertStores,
 	}
 
-	return notaryVerifier.New(&conf.TrustPolicyDoc, store, nil)
+	return notaryVerifier.New(&conf.TrustPolicyDoc, store, NewRatifyPluginManager(defaultPluginDirectory))
```

Comment:
Overriding the default is not a common scenario we have seen apart from during development where it's easier to switch the plugin directory to be in the dev workspace `bin/` folder

Reply:
yes, that is the intention. in that case, will update to use the configured directory instead, thanks!

Reply:
actually, upon closer inspection, it doesn't look like there's any good way to get the configured plugin directory from within the notaryv2 verifier, especially since each plugin could (in theory) be installed in a different directory. are you ok with requiring the user to install notation plugins in the default directory? or is there some central config that tracks where verifier plugins have been installed?

Reply:
I see the issue. Yeah the `pluginBinDirs` is not easily accessible from the notaryv2 verifier. Currently, all external plugins by default are installed in the same directory (even though the verifier config's `pluginBinDirs` is an array). See https://github.com/deislabs/ratify/blob/eeff667d7359ee2506eece29dc2799bcc1ed0b6a/pkg/verifier/config/config.go#L33
This value dictates the directory for all verifier plugins to be stored. If this value is overridden in the config, then I believe the default path currently used in the PluginManager constructor will not match. I don't have a preference on where the plugin gets downloaded to. It's just in its current form, any change to the pluginBinDirs field will result in a failure for the plugin manager.

Reply:
The ideal solution is to add the `pluginBinDirs` value as a parm for the verifier factory `Create` here: https://github.com/deislabs/ratify/blob/eeff667d7359ee2506eece29dc2799bcc1ed0b6a/pkg/verifier/factory/factory.go#L37 This issue doesn't apply to external verifiers because the external verifier factory already takes in a plugin path param: https://github.com/deislabs/ratify/blob/eeff667d7359ee2506eece29dc2799bcc1ed0b6a/pkg/verifier/factory/factory.go#L87

Reply:
I'm ok with us just filing an issue and revisiting if we can't get to this refactor in this PR. This might be a breaking change. @binbin-li any thoughts here on this?

Reply:
the change looks good to me. I'm Ok to address it in the following PRs.